stdlib/universe/typeconv.go

```diff
@@ -44,654 +43,298 @@ const (
 		}
-		str = sb.String()
-	default:
-		return nil, errors.Newf(codes.Invalid, "cannot convert %v to string", v.Type())
-	}
-	return values.NewString(str), nil
-}
-
-type intConv struct{}
-
-func (c *intConv) Type() semantic.MonoType {
-	return convIntType
-}
-
-func (c *intConv) IsNull() bool {
-	return false
-}
-func (c *intConv) Str() string {
-	panic(values.UnexpectedKind(semantic.Function, semantic.String))
-}
-func (c *intConv) Bytes() []byte {
-	panic(values.UnexpectedKind(semantic.Function, semantic.Bytes))
-}
-func (c *intConv) Int() int64 {
-	panic(values.UnexpectedKind(semantic.Function, semantic.Int))
-}
-func (c *intConv) UInt() uint64 {
-	panic(values.UnexpectedKind(semantic.Function, semantic.UInt))
-}
-func (c *intConv) Float() float64 {
-	panic(values.UnexpectedKind(semantic.Function, semantic.Float))
-}
-func (c *intConv) Bool() bool {
-	panic(values.UnexpectedKind(semantic.Function, semantic.Bool))
-}
-func (c *intConv) Time() values.Time {
-	panic(values.UnexpectedKind(semantic.Float, semantic.Time))
-}
-func (c *intConv) Duration() values.Duration {
-	panic(values.UnexpectedKind(semantic.Float, semantic.Duration))
-}
-func (c *intConv) Regexp() *regexp.Regexp {
-	panic(values.UnexpectedKind(semantic.Float, semantic.Regexp))
-}
-func (c *intConv) Array() values.Array {
-	panic(values.UnexpectedKind(semantic.Float, semantic.Array))
-}
-func (c *intConv) Object() values.Object {
-	panic(values.UnexpectedKind(semantic.Float, semantic.Object))
-}
-func (c *intConv) Function() values.Function {
-	return c
-}
-func (c *intConv) Dict() values.Dictionary {
-	panic(values.UnexpectedKind(semantic.Float, semantic.Dictionary))
-}
-func (c *intConv) Equal(rhs values.Value) bool {
-	f, ok := rhs.(*intConv)
-	return ok && (c == f)
-}
-func (c *intConv) HasSideEffect() bool {
-	return false
-}
+		return values.NewString(str), nil
+	},
+	false,
+)
 
-func (c *intConv) Call(ctx context.Context, args values.Object) (values.Value, error) {
-	var i int64
-	v, ok := args.Get(conversionArg)
-	if !ok {
-		return nil, errMissingArg
-	} else if v.IsNull() {
-		return values.Null, nil
-	}
-	switch v.Type().Nature() {
-	case semantic.String:
-		n, err := strconv.ParseInt(v.Str(), 10, 64)
-		if err != nil {
-			return nil, err
+var intConv = values.NewFunction(
+	"int",
+	convIntType,
+	func(ctx context.Context, args values.Object) (values.Value, error) {
+		var i int64
+		v, ok := args.Get(conversionArg)
+		if !ok {
+			return nil, errMissingArg
+		} else if v.IsNull() {
+			return values.Null, nil
 		}
-		i = n
-	case semantic.Int:
-		i = v.Int()
-	case semantic.UInt:
-		i = int64(v.UInt())
-	case semantic.Float:
-		i = int64(v.Float())
-	case semantic.Bool:
-		if v.Bool() {
-			i = 1
-		} else {
-			i = 0
+		switch v.Type().Nature() {
+		case semantic.String:
+			n, err := strconv.ParseInt(v.Str(), 10, 64)
+			if err != nil {
+				return nil, errors.Newf(codes.Invalid, "cannot convert string \"%v\" to int due to invalid syntax", v.Str())
```

Comment:
Go has a way to quote strings for you in formatting. This line can be 

```
return nil, errors.Newf(codes.Invalid, "cannot convert string %q to int due to invalid syntax", v.Str())
```

Note the `%q` instead of `\"%v\"`.

Reply:
Oh, I see you are using `%q` below, is there a reason its not being used here?

Reply:
I think it was just an oversight, good catch.